### PR TITLE
Reduce memory usage for closed ORC and Parquet writers

### PR DESCRIPTION
**Pull request by electrum**
## Release notes

(x) Release notes are required, with the following suggested text:

```markdown
# Delta Lake connector
* Reduce memory overhead for long running write queries. ({issue}`14824`)

# Hive connector
* Reduce memory overhead for long running write queries. ({issue}`14824`)

# Iceberg connector
* Reduce memory overhead for long running write queries. ({issue}`14824`)
```

### CODE REVIEW

File: lib/trino-orc/src/main/java/io/trino/orc/OrcWriter.java [hunk 1]
```diff
@@ -357,6 +355,15 @@ private void flushStripe(FlushReason flushReason)
         // write all data
         orcDataSink.write(outputData);
 
+        if (flushReason == CLOSED) {
```
Comment (sopel39) [inline lib/trino-orc/src/main/java/io/trino/orc/OrcWriter.java:358]: there is plenty of fields in both `OrcWriter` and `ParquetWriter`, (e.g `orcDataSink`). These are not freed by this code. Writers themselves seem are big objects so it doesn't make sense to keep them when they are not needed

Comment (electrum) [reply lib/trino-orc/src/main/java/io/trino/orc/OrcWriter.java:358]: Do any of these fields hold substantial memory?

Comment (sopel39) [reply lib/trino-orc/src/main/java/io/trino/orc/OrcWriter.java:358]: Yes

File: lib/trino-parquet/src/main/java/io/trino/parquet/writer/ParquetWriter.java [hunk 2]
```diff
@@ -216,6 +216,7 @@ public void close()
             flush();
             writeFooter();
         }
+        columnWriters.clear();
```
Comment (raunaqmorarka) [inline lib/trino-parquet/src/main/java/io/trino/parquet/writer/ParquetWriter.java:219]: FYI I had attempted similar change in https://github.com/trinodb/trino/commit/822f43e00c2db438291cbcbc4dac5c64cd426766
It resulted in a query failure (caught by benchmark suites outside of trino repo tests) and I had to revert it in https://github.com/trinodb/trino/pull/13263 (see error stack trace in the PR)
cc: @sopel39

**Comment by mosabua** [comment]
Can we close this @electrum ?

**Comment by raunaqmorarka** [comment]
This is already tackled by https://github.com/trinodb/trino/pull/14823

**Comment by mosabua** [comment]
Thank you @raunaqmorarka

**Pull request closed without merging**